### ISSUE TITLE
"Username exists" error page on last step of student registration via sign-up link

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-www#1284]
**TL;DR:** One report from a teacher that several of their students, seemingly randomly, hit an error page ("username exists") at the end of registration. However, the account is created afterwards.

**==== >^..^< ====**

@rschamp & I have been trying to figure out what could be going on with this user, but haven't been able to repro the issue or come up with an obvious theory. We have only gotten the one report of this issue or anything sounding like it so far, and normally that wouldn't be enough to warrant a case without some kind of repro steps or stacktrace, but this user is a teacher and has seen the issue with multiple students.

### Description of the issue from the user

From freshdesk no. 78663:
> When having my students go to my class join link (see ticket) and create accounts, many received an error that the user already exists for that username. This happened after entering their personal information. If the student navigated to the main scratch sign in page, they were able to login with the new account, even though they did not make it all the way through account creation and joining the class. This caused some students to think they needed to use a different username and caused them to create a different account. This caused duplicate student accounts in my class, with different usernames. 

Metadata:
- User: (see ticket)
- Operating System: Windows 7
- Browser: Chrome 57.0.2987.133
- Flash Version: Flash 25.0 (release 0)

> The students were seeing the duplicate unsername message after entering their person information.  It is the next screen after creating the username and password. 

> I ended the class and started a new class.  I had 2 students get the same error as yesterday, when joining the new class.  After entering their personal information, it said the user already exists.  At that point I had them exit the signup page, go to the main scratch website, and sign in with the newly created account.  The students then showed up in the class.   Here is the link to the new class: (see ticket)

> That is the error message we were seeing (in response to a screenshot from me).  However, these were for brand new accounts and I know for sure the students, today, did not have more than one tab open.  It seems to be random, as not all students saw the error.  I am working with the students to delete the accounts.

> It happened again today, in the same class, to these following user names: (see ticket). After entering their personal info and clicking the next button, they received the same error you sent in the screenshot. 

> The reason my students have duplicate accounts is, they saw the error that the username already exists after submitting the personal info. Not knowing that the user had actually been created, and that they could just go sign-in, they then went back to the signup link and tried a different username.  This caused them to end up with two usernames like: (see ticket)

**Some questions we asked the user:**

1) How are you asking them to register, are they typing the link into the browser navigation bar or opening it from their email or ...? 
> I supplied the signup link to the students on my Google Classroom page.

2) Where are they getting their username from? 
> The students are using (see ticket) to sign up. I ask them to use this number so they will not forget their username.  (Their username has never been entered on scratch site prior). 

3) Are they automatically logged in (to that account) after they get the error message? 
> No. The students must go to the scratch website to sign in with the newly created username and password.

4) Can they log into the account using the password they just supplied when they filled out the steps? Are they all creating their own passwords? 
> Yes and Yes.

5) How's your internet connectivity - would you describe it as slow by any chance? Very good Internet connectivity. 
> Not slow, no disconnection issue.

### Attempts at Repro (unsuccessful)
I created several test users (see ticket) using the sign-up link from the teacher and my own sign-up link on Staging, but have not been able to repro the issue. I tried to repro via:
- Throttling my internet speed (Chrome) and trying to click the last Next step button multiple times
- Refreshing the page partway through registration, then registering with the same username
- Using the back / forward arrows in the browser to move around registration steps
- Creating accounts with similar names
- Triggering the username validation then correcting the username to something else
- Filling out registration in the same tab as the error page was just triggered in
- **I did re-create the same error page the students saw, but only by filling out the same registration in two tabs simultaneously** (like, using the same username, and switching between tabs). This seems like the correct behaviour for that scenario, since only 1 of the accounts should be created with that username.

### The error page
The teacher confirmed that they were seeing the error page, not the validation message speech bubble on the username field.

The students were seeing this:
<img width="746" alt="screen shot 2017-04-24 at 10 52 22 am" src="https://cloud.githubusercontent.com/assets/1320456/25343186/4d4d49b0-28dc-11e7-8da0-e0d0aba28172.png">

...Not this:
<img width="585" alt="screen shot 2017-04-24 at 10 45 41 am" src="https://cloud.githubusercontent.com/assets/1320456/25343187/4d4efaf8-28dc-11e7-817f-f70326d4d13f.png">

**Comment by rschamp** [comment]
If we hear this behavior from another person, we should re-open.
